backend/app/server.py

```diff
@@ -44,7 +48,27 @@ async def ingest_files(
         if thread is None:
             raise HTTPException(status_code=404, detail="Thread not found.")
 
-    return ingest_runnable.batch([file.file for file in files], config)
+    if config["configurable"].get("show_progress_bar"):
```

Comment:
would maybe be better to branch based on request headers? eg the `Accept` header

Reply:
Using the `Accept` header here makes sense